Failed to get disk attachment list when using SSO authentication

When using SSO authentication, the following codes:
```go
func TestFollowLinks(t *testing.T) {
	inputRawURL := "https://10.1.111.229/ovirt-engine/api"

	conn, err := NewConnectionBuilder().
		URL(inputRawURL).
		Username("admin@internal").
		Password("qwer1234").
		Insecure(true).
		Compress(true).
		Timeout(time.Second * 10).
		Build()
	if err != nil {
		fmt.Printf("Make connection failed, reason: %v\n", err)
		return
	}
	defer conn.Close()

	vmsService := conn.SystemService().VmsService()

	// Error not checked is not recommended
	resp, err := vmsService.List().Search("name=test4joey").Send()
	if err != nil {
		fmt.Printf("Failed to search vm, reason: %v\n", err)
		return
	}
	vmSlice, _ := resp.Vms()
	vmService := vmsService.VmService(vmSlice.Slice()[0].MustId())
	dasService := vmService.DiskAttachmentsService()
	dasResp, err := dasService.List().Send()
	if err != nil {
		fmt.Printf("Failed to get das list, reason: %v\n", err)
		return
	}
	das, ok := dasResp.Attachments()  // This line causes the error
}
```

may generate the following errors: 
```go
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b12155]
```

When I did some debugs, the request URL of `dasService.List().Send()` is `https://10.1.111.229/ovirt-engine/sso/oauth/token/vms/becd686a-51e9-40fe-a24c-07a396841d93/diskattachments`, which should be `https://10.1.111.229/ovirt-engine/api/vms/becd686a-51e9-40fe-a24c-07a396841d93/diskattachments`.

--- a/sdk/ovirtsdk4/connection.go
+++ b/sdk/ovirtsdk4/connection.go
@@ -52,8 +52,6 @@ type Connection struct {
 	// SSO attributes
 	ssoToken     string
 	ssoTokenName string
-	ssoURL       *url.URL
-	ssoRevokeURL *url.URL
 }
 
 // URL returns the base URL of this connection.
@@ -303,11 +301,11 @@ func (c *Connection) buildSsoAuthRequest() (*url.URL, map[string]string) {
 	}
 
 	// Compute the URL:
-	ssoURL := c.url
+	var ssoURL url.URL = *c.url
 	ssoURL.Path = fmt.Sprintf("/ovirt-engine/sso/oauth/%s", entryPoint)
 
 	// Return the URL and the parameters:
-	return ssoURL, parameters
+	return &ssoURL, parameters
 }
 
 // buildSsoRevokeRequest builds a the URL and parameters to revoke the SSO access token.
@@ -321,11 +319,11 @@ func (c *Connection) buildSsoRevokeRequest() (*url.URL, map[string]string) {
 	}
 
 	// Compute the URL:
-	ssoURL := c.url
-	ssoURL.Path = "/ovirt-engine/services/sso-logout"
+	var ssoRevokeURL url.URL = *c.url
+	ssoRevokeURL.Path = "/ovirt-engine/services/sso-logout"
 
 	// Return the URL and the parameters:
-	return ssoURL, parameters
+	return &ssoRevokeURL, parameters
 }
 
 // SystemService returns a reference to the root of the services tree.

--- a/sdk/ovirtsdk4/service.go
+++ b/sdk/ovirtsdk4/service.go
@@ -56,15 +56,15 @@ func CheckFault(response *http.Response) error {
 	fault, err := XMLFaultReadOne(reader, nil, "")
 	if err != nil {
 		// If the XML is not a <fault>, just return nil
-		if _, ok := err.(XMLTagNotMatchError); ok {
-			return nil
+		if err, ok := err.(XMLTagNotMatchError); ok {
+			return err
 		}
 		return err
 	}
 	if fault != nil || response.StatusCode >= 400 {
 		return BuildError(response, fault)
 	}
-	return nil
+	return errors.New("unkonwn error")
 }
 
 // CheckAction checks if response contains an Action instance

